### ISSUE TITLE
Special character parsing error (&)

### ISSUE THREAD

**Issue by Laurens256** [Ulauncher/Ulauncher#1288]
### Environment

Mint 21.2 Victoria

### Bug description

Version 6.0.0 beta 1

I encountered a very strange glitch when searching for an application: when typing the letter `f`, my date & time settings item shows up. The & symbol is not parsed correctly and shows up as `&amp;`. ~~I haven't tested this but I suspect it may be similar for other special characters.~~
![20231024_172804](https://github.com/Ulauncher/Ulauncher/assets/85798751/08297def-f087-42e9-aea1-2e1bf91529ff)

The weird thing is, when searching for date and time regularly, the & symbol shows up as expected
![20231024_172816](https://github.com/Ulauncher/Ulauncher/assets/85798751/09b06720-6927-4299-8df2-f3549793f32d)

I'm not even sure date & time should show up at all, since there's no letter f in either the title or description. It doesn't show up for other letters either

(Sorry for pics of monitor btw)

### Log output
```
2023-10-24 17:37:07,201 ℹ️  INFO Ulauncher version 6.0.0-beta1 main.main:63
2023-10-24 17:37:07,201 ℹ️  INFO Extension API version 3.0 main.main:64
2023-10-24 17:37:07,201 ℹ️  INFO GTK+ 3.24.33 main.main:65
2023-10-24 17:37:07,201 ℹ️  INFO PyGObject+ 3.42.1 main.main:66
2023-10-24 17:37:07,201 ℹ️  INFO Desktop: X-CINNAMON (X11) on Linux Mint 21.2 main.main:68
2023-10-24 17:37:07,201 ⚠️  WARNING 

╔═════════════════════════════════════════════════════════════════════════════╗
║                  YOU ARE RUNNING A PRE-RELEASE of ULAUNCHER.                ║
║ Please do not report extension API support warnings to extension developers ║
║ We are still in the process of developing and documenting these features    ║
╚═════════════════════════════════════════════════════════════════════════════╝

 main.main:70
2023-10-24 17:37:07,201 ℹ️  INFO Migrating /home/laurens256/.config/ulauncher/ext_preferences/com.github.laurens256.ulauncher-random-number.json to /home/laurens256/.config/ulauncher/ext_preferences/com.github.laurens256.ulauncher-random-number.json migrate._migrate_file:44
2023-10-24 17:37:07,201 ℹ️  INFO Migrating /home/laurens256/.config/ulauncher/ext_preferences/com.github.manahter.ulauncher-translate.json to /home/laurens256/.config/ulauncher/ext_preferences/com.github.manahter.ulauncher-translate.json migrate._migrate_file:44
2023-10-24 17:37:07,201 ℹ️  INFO Migrating /home/laurens256/.config/ulauncher/ext_preferences/com.github.tchar.ulauncher-albert-calculate-anything.json to /home/laurens256/.config/ulauncher/ext_preferences/com.github.tchar.ulauncher-albert-calculate-anything.json migrate._migrate_file:44
2023-10-24 17:37:07,201 ℹ️  INFO Migrating /home/laurens256/.config/ulauncher/ext_preferences/com.github.brunoleitem.ulauncher-firefox.json to /home/laurens256/.config/ulauncher/ext_preferences/com.github.brunoleitem.ulauncher-firefox.json migrate._migrate_file:44
2023-10-24 17:37:07,202 ℹ️  INFO Migrating /home/laurens256/.config/ulauncher/ext_preferences/random-number.json to /home/laurens256/.config/ulauncher/ext_preferences/random-number.json migrate._migrate_file:44
2023-10-24 17:37:07,202 ℹ️  INFO Migrating /home/laurens256/.config/ulauncher/ext_preferences/com.github.the-lay.ulauncher-spotify-api.json to /home/laurens256/.config/ulauncher/ext_preferences/com.github.the-lay.ulauncher-spotify-api.json migrate._migrate_file:44
2023-10-24 17:37:07,202 ℹ️  INFO Migrating /home/laurens256/.config/ulauncher/ext_preferences/com.github.elken.terminal-runner.json to /home/laurens256/.config/ulauncher/ext_preferences/com.github.elken.terminal-runner.json migrate._migrate_file:44
2023-10-24 17:37:07,202 ℹ️  INFO Migrating /home/laurens256/.config/ulauncher/ext_preferences/com.github.sergius02.ulauncher-colorconverter.json to /home/laurens256/.config/ulauncher/ext_preferences/com.github.sergius02.ulauncher-colorconverter.json migrate._migrate_file:44
2023-10-24 17:37:07,202 ℹ️  INFO Migrating /home/laurens256/.config/ulauncher/ext_preferences/com.github.ulauncher.ulauncher-emoji.json to /home/laurens256/.config/ulauncher/ext_preferences/com.github.ulauncher.ulauncher-emoji.json migrate._migrate_file:44
2023-10-24 17:37:07,202 ℹ️  INFO Migrating /home/laurens256/.config/ulauncher/ext_preferences/com.github.nastuzzisamy.ulauncher-youtube-search.json to /home/laurens256/.config/ulauncher/ext_preferences/com.github.nastuzzisamy.ulauncher-youtube-search.json migrate._migrate_file:44
2023-10-24 17:37:07,202 ℹ️  INFO Migrating /home/laurens256/.config/ulauncher/ext_preferences/com.github.fajtak.ulauncher-weather.json to /home/laurens256/.config/ulauncher/ext_preferences/com.github.fajtak.ulauncher-weather.json migrate._migrate_file:44
2023-10-24 17:37:07,202 ℹ️  INFO Migrating /home/laurens256/.config/ulauncher/ext_preferences/com.github.brpaz.ulauncher-file-search.json to /home/laurens256/.config/ulauncher/ext_preferences/com.github.brpaz.ulauncher-file-search.json migrate._migrate_file:44
2023-10-24 17:37:07,202 ℹ️  INFO Migrating /home/laurens256/.config/ulauncher/ext_preferences/com.github.pywkm.ulauncher-spotify.json to /home/laurens256/.config/ulauncher/ext_preferences/com.github.pywkm.ulauncher-spotify.json migrate._migrate_file:44
2023-10-24 17:37:07,202 ℹ️  INFO Migrating /home/laurens256/.config/ulauncher/ext_preferences/com.github.fisadev.ulauncher-better-file-browser.json to /home/laurens256/.config/ulauncher/ext_preferences/com.github.fisadev.ulauncher-better-file-browser.json migrate._migrate_file:44
```

### Communication guidelines

- [X] I have read and agree with Ulauncher's communication guidelines and checked the troubleshooting page for self-help.

### Not an extension issue

- [X] This issue applies to the Ulauncher application, not a specific extension.

### Ulauncher v6

- [X] I have made sure the bug isn't already fixed in v6.

**Comment by friday** [comment]
Thanks! It's indeed a strange one, that is also hard to debug since search results are individual.

My guess is this has to do with the highlighter logic (that makes your search phrase blue in your results).

**Comment by Laurens256** [comment]
I think you're right, I can't actually test this but if I had to guess it's to do with this part of the code, specifically the `set_name` function:
https://github.com/Ulauncher/Ulauncher/blob/cfa2a779f3a54ff26a5532d913100d6627870d2f/ulauncher/ui/ResultWidget.py#L104C9-L122

If I'm reading this correctly, it only applies markup if a highlight `span` is found, but since there's no span because it isn't highlighted, it gets parsed as regular text

```py
  def set_name(self, name: str) -> None:
      item = self.builder.get_object("item-name")
      if "<span" in name:  # dealing with markup
          item.set_markup(name)
      else:
          item.set_text(name)
      self.name = name
```

**Comment by friday** [comment]
Why don't you try to check out the project locally and change that? You can find the development instructions from the readme.

**Comment by Laurens256** [comment]
Will do if I manage to get it working, I'm not really experienced with stuff like this but I'll give it a go :) thanks

**Comment by friday** [comment]
Should be fixed now by 1293 and e62850e0a3142cb3b51d7761c1d0371fc97da6bc right? Let me knowotherwise
